.github/testflinger/job-def.yaml

```diff
@@ -0,0 +1,116 @@
+job_queue: dell-precision-3470-c30322
+provision_data:
+  distro: jammy
+  user_data: |
+    #cloud-config
+    # Note, much of this is borrowed from the
+    # charm-dev multipass template:
+    # https://github.com/canonical/multipass-blueprints/blob/main/v1/charm-dev.yaml
+
+    runcmd:
+    - |
+      # Install pip from apt here instead of in packages
+      # section to ensure process is sequential
+      DEBIAN_FRONTEND=noninteractive apt update
+      DEBIAN_FRONTEND=noninteractive apt -y upgrade
+      DEBIAN_FRONTEND=noninteractive apt -y install python3-pip
+
+      # Juju 3.5 is supported until Jan 2025
+      # https://juju.is/docs/juju/roadmap
+      snap install juju --channel=3.5/stable
+      snap install microk8s --channel=1.29-strict/stable
+      snap install --classic charmcraft
+      snap refresh
+
+    - |
+      # Install tox from pypi (v4) instead of apt (v3)
+      pip install tox
+
+    - |
+      # Disable swap
+      sysctl -w vm.swappiness=0
+      echo "vm.swappiness = 0" | tee -a /etc/sysctl.conf
+      swapoff -a
+
+      # Disable IPv6
+      echo "net.ipv6.conf.all.disable_ipv6=1" | tee -a /etc/sysctl.conf
+      echo "net.ipv6.conf.default.disable_ipv6=1" | tee -a /etc/sysctl.conf
+      echo "net.ipv6.conf.lo.disable_ipv6=1" | tee -a /etc/sysctl.conf
+      sysctl -p
+
+    - |
+      # Make sure juju directory is there
+      # https://bugs.launchpad.net/juju/+bug/1995697
+      sudo -u ubuntu mkdir -p /home/ubuntu/.local/share/juju
+
+    - |
+      # setup lxd for charmcraft
+      lxd init --auto
+      adduser ubuntu lxd
+
+    - |
+      # setup microk8s and bootstrap
+      usermod -a -G snap_microk8s ubuntu
+      usermod -a -G microk8s ubuntu
+      microk8s status --wait-ready
+      microk8s.enable dns
+      microk8s.kubectl rollout status \
+        deployments/coredns \
+        -n kube-system -w --timeout=10800s
+      microk8s.enable hostpath-storage
+      # wait for storage to become available
+      microk8s.kubectl rollout status \
+        deployments/hostpath-provisioner \
+        -n kube-system -w --timeout=10800s
+      sudo -u ubuntu juju bootstrap microk8s microk8s
+      sudo -u ubuntu juju add-model \
+        --config logging-config="<root>=WARNING; unit=DEBUG" \
+        --config update-status-hook-interval="1m" \
+        welcome-k8s
+test_data:
+  test_cmds: |
+
+    # Exit immediately if a test fails
+    set -e
+
+    # Clone repo from appropriate branch/commit.
+    # This is done here rather than in user_data to avoid
+    # race conditions were the test_cmds section runs
+    # before the runcmd section has run on the initial boot.
+    ssh $DEVICE_IP '
```

Comment:
ssh $DEVICE_IP is enough ?
- adding ubuntu user would be nice to have ?
- do we need ssh options ? i always had to use `-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null` so a little bit surprised it works fine for you

Reply:
I will add `ubuntu@$DEVICE_ID` - I agree it's nice for the user to be shown more explicitly.

On the other ssh options, I'm guessing the testflinger server (where the `ssh` command is issue) either has all the device host keys installed or it's configured to ignore host keys. If you use `reserve_data` and are trying to connect from your personal machine then you may want those options if you don't want to remove the host keys between each re-provision.

Reply:
yeah; probably, i see these options used anywhere but maybe it is not necessary anymore; for example; this is the ssh helper function cert team use in the TF job for SRU testing : https://github.com/canonical/hwcert-jenkins-jobs/blob/cd3c60a045fb059084a20f47ce475a49a4477b9d/jobs/sru/test-cmds-convenience-functions.sh#L12